.github/workflows/comment_on_first_contribution.yml

```
@@ -0,0 +1,56 @@
+name: Comment on first contribution
+
+on:
+  workflow_call:
+    inputs:
+      pull_request_message:
+        description: >
+          The Markdown message to comment when a contributor creates their first
+          pull request.
+        type: string
+        required: false
+      issue_message:
+        description: >
+          The Markdown message to comment when a contributor creates their first
+          issue.
+        type: string
+        required: false
+
+jobs:
+  comment:
+    if: >
+      (
+        (github.event_name == 'pull_request_target' &&
+         inputs.pull_request_message) ||
+        (github.event_name == 'issues' && inputs.issue_message)
+      ) &&
+      github.event.action == 'opened' &&
+      contains(
+        fromJSON('["FIRST_TIMER", "FIRST_TIME_CONTRIBUTOR"]'),
```

Is `fromJSON` needed here or could this just be the following?

```suggestion
        ["FIRST_TIMER", "FIRST_TIME_CONTRIBUTOR"],
```

---

There appears to be no better way to [construct an array](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/evaluate-expressions-in-workflows-and-actions#example-matching-an-array-of-strings). No array literals either.

---

Oh, wow. I didn’t realize that. Fair enough then.